src/components/document/document-content.tsx

```diff
@@ -53,25 +54,41 @@ export class DocumentContentComponent extends BaseComponent<IProps, IState> {
   private domElement: HTMLElement | null;
   private rowRefs: Array<TileRowComponent | null>;
   private mutationObserver: MutationObserver;
+  private scrollDisposer: IReactionDisposer;
 
   public componentDidMount() {
     if (this.domElement) {
       this.domElement.addEventListener("scroll", throttle(this.updateVisibleRows, 100));
       this.updateVisibleRows();
       this.scrollToSection(this.props.selectedSectionId);
 
-      if ((window as any).MutationObserver) {
-        this.mutationObserver = new MutationObserver(this.handleRowElementsChanged);
-        this.mutationObserver.observe(this.domElement, { childList: true });
-      }
       // We pass the domElement to our children, but it's undefined during the first render,
       // so we force an update to make sure we draw at least once after we have our domElement.
       this.forceUpdate();
+
+      this.scrollDisposer = reaction(() => this.stores.ui.scrollToTileId,
+        (tileId: string | undefined, prevTileId: string | undefined) => {
+        if (tileId) {
+          this.rowRefs.forEach((row: TileRowComponent | null) => {
+            if (row?.tileRowDiv && row.hasTile(tileId)) {
+              // Javascript struggles to scroll multiple elements at the same time,
+              // so we delay scrolling any document on the left and only animate the left document
+              setTimeout(() => {
+                row?.tileRowDiv?.scrollIntoView({
+                  behavior: this.props.readOnly ? "smooth" : "auto",
+                  block: "nearest",
+                  inline: "nearest"
+                });
+              }, this.props.readOnly ? 100 : 1);
+            }
+          });
+        }
+      });
     }
   }
 
   public componentWillUnmount() {
-    this.mutationObserver.disconnect();
+    this.scrollDisposer?.();
```

Comment:
Newbie question about reactions: How does this dispose/end the reaction? Is that when you setup the reaction above, what you get back is the function to call when you're done paying attention?  I also realize, I don't understand what the `.` before the `()`does

Reply:
Yes, a mobx `reaction` or `autorun` returns a function that cleans the listener up when you call it.

The `.` is weird syntax that allows you to call a function after using `?` to check to make sure it exists. That line is the equivalent of `if (this.scrollDisposer) this.scrollDisposer();` or `this.scrollDisposer && this.scrollDisposer();`.